data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaTopicProducerMetrics.java

```diff
@@ -0,0 +1,129 @@
+                pluginMetrics.gauge(getTopicMetricName(camelCaseName, topicNameInMetrics), metricValues, metricValues -> {
+                    double sum = 0;
+                    for (Map.Entry<KafkaProducer, Map<String, Double>> entry : metricValues.entrySet()) {
+                        Map<String, Double> producerMetrics = entry.getValue();
+                        synchronized(producerMetrics) {
+                            sum += producerMetrics.get(metricName);
+                        }
+                    }
+                    return sum;
+                });
+            }
+        });
+    }
+
+    public void register(final KafkaProducer producer) {
+        metricValues.put(producer, new HashMap<>());
+        final Map<String, Double> producerMetrics = metricValues.get(producer);
+        metricsNameMap.forEach((k, name) -> {
+            producerMetrics.put(k, 0.0);
+        });
+    }
+
+    Counter getNumberOfRecordsSent() {
+        return numberOfRecordsSent;
+    }
+
+    Counter getNumberOfBytesSent() {
+        return numberOfBytesSent;
+    }
+
+    private String getTopicMetricName(final String metricName, final boolean topicNameInMetrics) {
+        if (topicNameInMetrics) {
+            return "topic." + topicName + "." + metricName;
+        } else {
+            return metricName;
+        }
+    }
+
+    Map<KafkaProducer, Map<String, Double>> getMetricValues() {
+        return metricValues;
+    }
+
+    public void update(final KafkaProducer producer) {
+        Map<String, Double> producerMetrics = metricValues.get(producer);
+
+        Map<MetricName, ? extends Metric> metrics = producer.metrics();
+        for (Map.Entry<MetricName, ? extends Metric> entry : metrics.entrySet()) {
+            MetricName metric = entry.getKey();
+            Metric value = entry.getValue();
+            String metricName = metric.name();
+            //if(metric.tags().containsKey("topic") && metricName.contains("lag")) {
+            if ((metricName.contains("send") || metricName.contains("byte")) && metric.tags().containsKey("topic")) {
+                System.out.println("Metric Name: " + metricName);
+                System.out.println("Metric Value: " + value.metricValue());
+                System.out.println("Metric Tag: " + metric.tags());
+            }
+            if (Objects.nonNull(metricsNameMap.get(metricName))) {
+                // producer metrics are emitted at topic level
+                if (!metric.tags().containsKey("topic")) {
+                   continue;
+                }
+
+                double newValue = (Double)value.metricValue();
+                if (metricName.equals("record-send-total")) {
+                    synchronized(producerMetrics) {
+                        double prevValue = producerMetrics.get(metricName);
+                        numberOfRecordsSent.increment(newValue - prevValue);
+                    }
+                } else if (metricName.equals("byte-total")) {
```

Comment:
Lot of these keyword strings are used multiple times. Would be better to put them into variables.

Reply:
Will fix.